src/nbl/ui/CWindowX11.cpp

```diff
@@ -301,15 +310,53 @@ void CWindowX11::processEvent(XEvent event)
         // call mouse/keyboard/both focus change 
         case FocusIn:
         {
+            
             break;
         }
         case FocusOut:
         {
+            
             break;
         }
     }
 }
 
+void CWindowManagerX11::init()
```

Comment:
i believe those are methods of thread handler, not CWindowManagerX11's

Reply:
isn't X11 safe to use in threads? with `XinitThreads` ?

Reply:
do you even need a thread handler for X11 ?

Reply:
for events yes, ofc. For window creation/destruction im not sure it's mandatory like in case of Windows, so i assumed yes.

XNextEvent is blocking until an event appears so in case of of window creation/destruction request, we need to generate a dummy event to wake it up

Reply:
> XNextEvent is blocking until an event appears so in case of of window creation/destruction request, we need to generate a dummy event to wake it up

you only need a dedicated thread to poll events then, nothing else (and a dummy event to make the loop quit)

Reply:
are you sure? on Windows the thread polling events and the thread creating windows must be the same thread. Im not sure it's not the case here

Reply:
this is X11 we're talking about

Reply:
I dont think you need a threadhandler for X11